interlay/interbtc
# feat: POC of contracts pallet

This adds pallet_contracts to the `standalone` runtime. I added a contract in `contracts/hello_world` that has a function that interacts with the runtime, and a test in `standalone/runtime/tests/test_contracts.rs` to show that it's working. The test is simple: the contract calls a function in the runtime that returns some value (in this case, it returns Alice's balance), followed by another call that sets some value. This way, we test that (1) passing arguments to runtime functions work, (2) those functions can return values, and (3) that these functions can have side effects (storage changes). The implementation of the runtime side of these functions can be found in `standalone/runtime/src/lib.rs`

To test for yourself: 

- run `cargo install cargo-contract`
- `cd` to the `contracts/hello_world` folder and run `cargo contract build`
- open `standalone/runtime/tests/test_contracts.rs`
  - Modify the path in `test_contract`
  - run `test_contract`
  - Observe that the test passes

ps don't mind the dex changes - they are unrelated

Closes https://github.com/interlay/interbtc/issues/983
